netflix/hystrix/index.js

```diff
@@ -9,51 +10,58 @@ exports.setHystrixMetricsStreamHandlerFactory = function(handler){
 	hystrixMetricsStreamHandlerFactory= handler;
 }
 
-exports.hystrixStream = function(request, response) {
+exports.hystrixStream = function(request, reply) {
 	
 	if (!hystrixMetricsStreamHandlerFactory)
 	{
-		response.send(503,{error:"hystrixMetricsStreamHandlerFactory not defined."});
+		reply.code(503).send({error:"hystrixMetricsStreamHandlerFactory not defined."});
 		return;
 	}
 
 	console.log('setup hystrix stream with:' +refreshInterval +" ms");
 	
 	hystrixMetricsStreamHandlerFactory.getHystrixMetricsStreamHandler(refreshInterval, function(error, instance){
-		if (!instance ){ 
+		console.log('getHystrixMetricsStreamHandler')
+
+		if (!instance) { 
 			console.log("Can not get instance");
-			response.send(503, {error: 'Can not get instance'});
-        }else if (error){ 
-            console.log("Get instance hit error:"+error);
-            response.send(503, {error: error});
-        }else {
-             //End is never called now. Where else I can shutdown the instances which will the connection count???
-			 request.on('end', function() {
-				 console.log('receive request end');
-                 instance.shutdown(function(){      }) 
-			 });
-            /* initialize response */
-            response.setHeader("Content-Type", "text/event-stream;charset=UTF-8");
-            response.setHeader("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
-            response.setHeader("Pragma", "no-cache");
-
- 		    setInterval(function(){
-			    instance.getJsonMessageAsString(function(err,jsonMessageStr){
-				    if (err) {
-				    	console.log("error:"+err);
-						response.send(503,{error: err});
-	                } else  {
+			reply.code(503).send({error: 'Can not get instance'});
+		} else if (error) { 
+			console.log("Get instance hit error:"+error);
```

Comment:
maybe add space after `:` so the error message does not get strictly concatenated to the initial message

Reply:
I haven't really focused on this code path.  If it's necessary to move forward I'll bring in your ideas.